### PR TITLE
Upload User and Project files; Getting file urls in api responses; integrating Eloquent API resources

### PR DESCRIPTION
**Pull request by DanielKiel**
see https://laravel.com/docs/7.x/eloquent-resources
before implementing the connection to the frontend, this minor change in API responses will be necessary to have control of the api responses structures.
Changes are documented in API_DOCS

Integrated media library https://docs.spatie.be/laravel-medialibrary/v7/introduction/ is now API ready to upload files at User and at Project. It can handle images but also other file types.

### CODE REVIEW

File: app/Http/Resources/UserResource.php [hunk 1]
```diff
@@ -14,6 +15,29 @@ class UserResource extends JsonResource
...
+    }
+
+    /**
+     * @return Collection
+     */
+    public function getFileURLs(): Collection
+    {
+        $urls = collect();
+
+        $this->getMedia('project-files')->each(function($media) use($urls) {
```
Comment (comes) [inline app/Http/Resources/UserResource.php]: ich nutz hier in zwischen lieber map()

```
return $this->getMedia('project-files')->map(function($media) {
  return $media->getURL();
});
```

File: app/Http/Resources/ProjectResource.php [hunk 2]
```diff
@@ -16,13 +17,30 @@ class ProjectResource extends JsonResource
...
+    /**
+     * @return Collection
+     */
+    public function getFileURLs(): Collection
+    {
+        $urls = collect();
+
+        $this->getMedia('project-files')->each(function($media) use($urls) {
+            $urls->push($media->getURL());
+        });
```
Comment (comes) [inline app/Http/Resources/ProjectResource.php]:
```
return $this->getMedia('project-files')->map(function($media) {
  return $media->getURL();
});
```

Comment (DanielKiel) [reply app/Http/Resources/ProjectResource.php]: werd ich mir mal versuchen zu merken ... each() ist ne alte gewohnheit

File: app/Http/Resources/UserResource.php (hunk 1 above)
Comment (DanielKiel) [reply app/Http/Resources/UserResource.php]: werd ich mir mal versuchen zu merken ... each() ist ne alte gewohnheit

**Pull request merged**